feature-libs/organization/user-registration/components/form/user-registration-form.component.ts

```diff
@@ -1,42 +1,39 @@
-import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { ChangeDetectionStrategy, Component, OnDestroy } from '@angular/core';
 import { FormGroup } from '@angular/forms';
 import { Country, Region } from '@spartacus/core';
 import { Title } from '@spartacus/user/profile/root';
-import { BehaviorSubject, Observable } from 'rxjs';
+import { BehaviorSubject, Observable, Subscription } from 'rxjs';
 import { UserRegistrationFormService } from './user-registration-form.service';
 
 @Component({
   selector: 'cx-user-registration-form',
   templateUrl: './user-registration-form.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class UserRegistrationFormComponent implements OnInit {
-  titles$: Observable<Title[]>;
+export class UserRegistrationFormComponent implements OnDestroy {
+  titles$: Observable<Title[]> = this.userRegistrationFormService.getTitles();
 
-  countries$: Observable<Country[]>;
+  countries$: Observable<Country[]> =
+    this.userRegistrationFormService.getCountries();
 
-  regions$: Observable<Region[]>;
+  regions$: Observable<Region[]> =
+    this.userRegistrationFormService.getRegions();
+
+  registerForm: FormGroup = this.userRegistrationFormService.form;
 
   isLoading$ = new BehaviorSubject(false);
 
-  registerForm: FormGroup;
+  protected registrationSub = new Subscription();
```

Comment:
[nitpick] Let's not limit the subscriptions to one per stream, and let's use a generic name for all potential subscriptions in the future - I would suggest to use just `subscriptions` as a name

Reply:
And nice that you marked it as `protected`

Reply:
Done.